locust/distribution.py

```diff
@@ -0,0 +1,135 @@
+import math
+from itertools import combinations_with_replacement
+from operator import attrgetter
+from typing import (
+    Dict,
+    List,
+    Type,
+)
+
+from locust import User
+
+
+def weight_users(
```

Comment:
How about something like `compute_users_distribution` or `compute_desired_users_distribution`? Let me know what makes more sense. I can also simply `user_weights`, but it's not really what the function does.

Reply:
Hmm... I think maybe this is a clash between OO and functional naming styles. In OO, functions are named either for what they accomplish (side effects) or what they return (getters or other methods that look at the internal state of the object). 

In OO a function called "weigh_users" sounds like it is adding weights (side effects) to users. "compute_user_distribution" makes sense, but "compute" is redundant, as most methods with return values do some sort of computation :) 

Maybe user_distribution makes sense?

Another thing that got me thinking is why do you return and then pass around class names instead of classes? Type-hinted method signatures would be much nicer if we could just use classes, but maybe there is good reason for it?

Reply:
There's also the (perhaps unlikely, but super sneaky if it were to happen) case of having multiple classes named the same thing, but from different places (like foo.MyUser, bar.MyUser). Not sure that is really possible but just the thought of debugging something like that scares me :P